core/include/detray/propagator/actors/parameter_transporter.hpp

```diff
@@ -153,11 +139,34 @@ struct parameter_transporter : actor {
                     prev_sf.transform(ctx), propagation);
         }
 
-        sf.template visit_mask<kernel>(sf.transform(ctx),
-                                       bound_to_free_jacobian, propagation);
+        return sf.template visit_mask<full_jacobian_kernel>(
+            sf.transform(ctx), bound_to_free_jacobian, propagation);
```

Comment:
The previous surface does not exist only happens when it starts from free track parameter. And I am assuming that we never do covariance transport for this case. (i.e. Input with free parameters is only for validation & simulation)

Actually the covariance transport of this case has been implicitly deprecated since #790. I am putting a dummy identity matrix to the bound covariance 

```
        /// Sets track parameters.
        DETRAY_HOST_DEVICE
        explicit state(const free_track_parameters_type &free_params)
            : _track(free_params) {

            curvilinear_frame<algebra_t> cf(free_params);

            // Set bound track parameters
            _bound_params.set_parameter_vector(cf.m_bound_vec);

            // A dummy covariance - should not be used
            _bound_params.set_covariance(
                matrix_operator()
                    .template identity<e_bound_size, e_bound_size>());

            // A dummy barcode - should not be used
            _bound_params.set_surface_link(geometry::barcode{});

            // Reset the path length
            _s = 0.f;

            // Reset jacobian transport to identity matrix
            matrix_operator().set_identity(_jac_transport);
        }
```

Having said that, the covariance transport in benchmarks is ill-defined currently

Reply:
Ah, because the parameter resetter is called in the base stepper state?

Reply:
Yes. the parameter resetter is called in the base stepper state with bound paramter input, which sets the _prev_sf_id to valide value.

Reply:
Ok, I see, thanks!

Reply:
You could add the free covariance as an additional parameter to the stepper constructor and then it would be up to the caller to pass on something valid or not

Reply:
That is probably what we have done in the past.
I disabled it because I thought it is cumbersome to generate a well-defined free covariance to be used for covariance transport.

Anyway we can follow up this later.